File: backend/src/models/connection.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum DbType {
    Postgres,
    Mysql,
    Sqlite,
}

#[derive(Debug, Deserialize)]
pub struct ConnectRequest {
    pub host: String,
    pub port: u16,
    pub database: String,
    pub username: String,
    pub password: String,
    pub db_type: DbType,
}

#[derive(Debug, Serialize)]
pub struct ConnectResponse {
    pub token: String,
    pub database: String,
    pub db_type: DbType,
}

#[derive(Debug, Serialize)]
pub struct StatusResponse {
    pub connected: bool,
    pub database: Option<String>,
    pub db_type: Option<DbType>,
}

File: ui/src/models/connection.ts
// Database connection types

export type DatabaseType = 'postgres' | 'mysql' | 'sqlite';

export interface ConnectionForm {
    dbType: DatabaseType;
    database: string;
    host: string;
    port: string;
    username: string;
    password: string;
}

export const DEFAULT_PORTS: Record<DatabaseType, string> = {
    postgres: '5432',
    mysql: '3306',
    sqlite: '',
};

export const INITIAL_CONNECTION_FORM: ConnectionForm = {
    dbType: 'postgres',
    database: '',
    host: 'localhost',
    port: '5432',
    username: '',
    password: '',
};
